### ISSUE TITLE
CMD+` seems to linger thinking for a second or two when window circling is completed.

### ISSUE THREAD

**Issue by krstp** [zed-industries/zed#7416]
### Check for existing issues

- [X] Completed

### Describe the feature

When I use CMD+` (circle between windows), ZED seems to get stuck for a second of two causing visible hold before jumping back to the first window. This seems a more recent issue, as before I did not experience it.

### If applicable, add mockups / screenshots to help present your vision of the feature

```
Version: Zed Preview 0.121.3
Chip: Apple M2 Max
Memory: 32GB
macOS: Sonoma 14.2.1
```

**Comment by mrnugget** [comment]
Just to leave first information.

It happens when one Zed window is maximized and one isn't. When then flipping between them with <kbd>Cmd-`</kbd> it can cause a hang. I captured it in Instruments.app.

![screenshot-2024-02-06-11 04 38@2x](https://github.com/zed-industries/zed/assets/1185253/03ae3135-74a9-4e62-a068-410c7afa0589)

Turns out our `main` thread is blocked:

![screenshot-2024-02-06-11 16 29@2x](https://github.com/zed-industries/zed/assets/1185253/c3a232d1-7e79-449f-8034-ed4b18c2e04b)

According to the narrative view in `Instruments.app` it's a timed semaphore:
![screenshot-2024-02-06-11 17 12@2x](https://github.com/zed-industries/zed/assets/1185253/6b075ece-c622-4625-a51b-8373dba580c8)

**Comment by mrnugget** [comment]
According to this, it seems like we might have to stop/restart our displaylink when our window is in transition: https://github.com/gfx-rs/gfx/issues/2460#issuecomment-1550124134

**Comment by krstp** [comment]
FYI: In my case I don't need to have full screen, but I work on 4k size monitor, so might be directly related to the size of the window, which for one of those takes up majority of the screen. Thanks for checking that.

**Comment by as-cii** [comment]
This should be fixed in Zed Preview 0.121.4. Thanks for there report @krstp!
